MS Teams Integration (#3467)
owncloud/docs: modules/admin_manual/pages/configuration/user/oidc/oidc.adoc

== Set Up Service Discovery

. Webserver Service Discovery Information
+
In order to allow the ownCloud Clients (Desktop/Android/iOS) to make use of OpenID Connect, the webserver serving ownCloud Server needs to _provide service discovery information_ under the following static path:
+
[source,url]
----
https://cloud.example.com/.well-known/openid-configuration
----

. App Service Discovery Information
+
When enabled, the OpenID Connect App provides the service discovery information on the endpoint:
+
[source,url]
----
https://cloud.example.com/index.php/apps/openidconnect/config
----

. Webserver Rewrite Rule
+
To make the endpoint available under the static service discovery path, it is recommended to put a `RewriteRule` in place using `.htaccess` (the Apache modules `proxy` and `proxy_http` have to be enabled):
+
[source,apache]
----
RewriteRule ^\.well-known/openid-configuration /index.php/apps/openidconnect/config [P]
----
+
TIP: Depending on the respective infrastructure setup there can be other ways to solve this. In any case, please make sure _not to use redirect rules_ as this will violate the OpenID Connect specification.
+

> IMPORTANT: For the time being and in case you use the `.htaccess` file in the ownCloud web root, you have to manually add that header again after any ownCloud upgrade.

Reviewer: ```suggestion
IMPORTANT: If you use the `.htaccess` file in the ownCloud web root, you have to manually add that rewrite rule again after any ownCloud upgrade.
```
